packages/zowe-explorer-ftp-extension/src/ZoweExplorerFtpUssApi.ts

```diff
@@ -81,16 +81,9 @@ export class FtpUssApi extends AbstractFtpApi implements ZoweExplorerApi.IUss {
         }
     }
 
-    public async putContents(
-        inputFilePath: string,
-        ussFilePath: string,
-        binary?: boolean,
-        localEncoding?: string,
-        etag?: string,
-        returnEtag?: boolean
-    ): Promise<zowe.IZosFilesResponse> {
+    public async putContent(inputFilePath: string, ussFilePath: string, options?: zowe.IUploadOptions): Promise<zowe.IZosFilesResponse> {
         const transferOptions = {
-            transferType: binary ? TRANSFER_TYPE_BINARY : TRANSFER_TYPE_ASCII,
+            transferType: options.binary ? TRANSFER_TYPE_BINARY : TRANSFER_TYPE_ASCII,
```

Comment:
This will throw an error if `options` parameter is undefined. We should either use `options?.binary` or default the parameter in the function signature to an empty object: `options: zowe.IUploadOptions = {}`

The same comment applies for `options.returnEtag` and `options.etag` below.

Reply:
I'm curious if all of this is duplication of code between our FTP Extension and the FTP Plug-in.
https://github.com/zowe/zowe-cli-ftp-plugin/blob/ze-copy-ds/src/api/DataSetUtils.ts#L286-L297